File: scala/org/softlang/company/Employee.scala
package org.softlang.company

/**
 * An Employee has some person information and receives a salary.
 */
class Employee {
  private var person: Person = _
  private var salary: Double = 0.0

  def getPerson = person
  def setPerson(person: Person) = {
    this.person = person
  }

  def getSalary = salary
  def setSalary(salary: Double) = {
    this.salary = salary
  }

  def total: Double = getSalary

  def cut: Unit = {
    setSalary(getSalary / 2)
  }

}

File: gwt/src/org/softlang/server/company/Employee.java
package org.softlang.server.company;

import java.io.Serializable;

/**
 * An Employee has a salary and some person information
 * 
 */
public class Employee implements Serializable {

	private static final long serialVersionUID = -210889592677165250L;

	private Person person;
	private double salary;

	public Person getPerson() {
		return person;
	}

	public void setPerson(Person person) {
		this.person = person;
	}

	public double getSalary() {
		return salary;
	}

	public void setSalary(double salary) {
		this.salary = salary;
	}

	public double total() {
		return getSalary();
	}

	public void cut() {
		setSalary(getSalary() / 2);
	}
}
